packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,348 @@
+					registerId,
+				);
+			await registerRoot.write("B", handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			const registerRoot =
+				await defaultDataStore.getSharedObject<IConsensusRegisterCollection<FluidObject>>(
+					registerId,
+				);
+			return registerRoot.read("B");
+		},
+	},
+	{
+		type: queueId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const queueRoot =
+				await defaultDataStore.getSharedObject<IConsensusOrderedCollection>(queueId);
+			await queueRoot.add(handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			let handle2: unknown;
+			const queueRoot =
+				await defaultDataStore.getSharedObject<IConsensusOrderedCollection>(queueId);
+			const callback: ConsensusCallback<IFluidHandle> = async (value) => {
+				handle2 = value;
+				return ConsensusResult.Complete;
+			};
+			await queueRoot.waitAndAcquire(callback);
+			return handle2;
+		},
+	},
+	// {
+	// 	type: migrationShimId,
+	// 	storeHandle: async (defaultDataStore, handle) => {
+	// 		const migrationShimRoot = await defaultDataStore.getSharedObject<MigrationShim>(migrationShimId);
+	// 		const tree = migrationShimRoot.currentTree as LegacySharedTree;
+	// 		const legacyNodeId: TraitLabel = "inventory" as TraitLabel;
+
+	// 		const handleNode: BuildNode = {
+	// 			definition: legacyNodeId,
+	// 			traits: {
+	// 				handle,
+	// 			},
+	// 		};
+	// 		tree.applyEdit(
+	// 			Change.insertTree(
+	// 				handleNode,
+	// 				StablePlace.atStartOf({
+	// 					parent: tree.currentView.root,
+	// 					label: legacyNodeId,
+	// 				}),
+	// 			),
+	// 		);
+
+	// 		const rootNode = tree.currentView.getViewNode(tree.currentView.root);
+	// 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	// 		const nodeId = rootNode.traits.get(legacyNodeId)![0];
+	// 		const change = Change.setPayload(nodeId, handle);
+	// 		tree.applyEdit(change);
+	// 	},
+	// },
+];
+
+describeCompat("handle validation", "NoCompat", (getTestObjectProvider, apis) => {
+	for (const handle of handleFns) {
+		it(`store handle in dds: ${handle.type}`, async () => {
+			const { container1, provider, testContainerConfig } = await setup(
+				getTestObjectProvider,
+				apis,
+			);
+			let idA: string;
+			let seq: number;
+			{
+				const defaultDataStore = (await container1.getEntryPoint()) as ITestFluidObject;
+				const runtime = defaultDataStore.context.containerRuntime;
+				idA = defaultDataStore.context.id;
+
+				const dataStoreB = await runtime.createDataStore(["default"]);
+				const dataObjectB = (await dataStoreB.entryPoint.get()) as ITestFluidObject;
+
+				await handle.storeHandle(defaultDataStore, dataObjectB.handle);
+
+				await provider.ensureSynchronized();
+				seq = container1.deltaManager.lastSequenceNumber;
+				container1.dispose();
+			}
+
+			const container2 = await provider.loadTestContainer(testContainerConfig);
+			if (container2.deltaManager.lastSequenceNumber < seq) {
```

Comment:
FYI this code already exists here:  https://github.com/microsoft/FluidFramework/tree/main/packages/loader/container-loader/src/catchUpMonitor.ts

Reply:
Hmm, almost... it doesn't support passing a custom target in.  Oh well.

Reply:
But maybe if you do provider.ensuresynchronized before loading container2, it would work